relayer/src/chain/handle/cache.rs

```diff
@@ -131,8 +132,15 @@ impl<Handle: ChainHandle> ChainHandle for CachingChainHandle<Handle> {
 
     fn query_latest_height(&self) -> Result<Height, Error> {
         let handle = self.inner();
-        self.cache
-            .get_or_try_update_latest_height_with(|| handle.query_latest_height())
+        let (result, in_cache) = self
+            .cache
+            .get_or_try_update_latest_height_with(|| handle.query_latest_height())?;
+
+        if in_cache == CacheStatus::Hit {
+            telemetry!(query_cache_hit, &self.id(), "query_latest_height");
```

Comment:
I guess the difficulty of pushing `telemetry!` calls inside the `cache` is that we don't have access to the chain identifier (`self.id()`). But maybe we can store the chain identifier as a field in the cache layer to fix this difficulty.